### PR TITLE
Feat: Add erase DB feature 

### PR DESCRIPTION
**Pull request by micache**
## 🎯 Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request._

_If it fixes a bug or resolves a feature request, be sure to link to that issue._

## ☑️ Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR._
- [x] I have added Labels for this PR
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective, or my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## ✍ Further comments

_If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc..._

### CODE REVIEW

**Comment by Huy1711** [comment]
please do git rebase first

File: app/llama_sensei/backend/add_courses/embedding/get_embedding.py [hunk 1]
```diff
@@ -24,6 +24,6 @@ def embed_chunks(self, chunks: List[tuple], top_chunks: int = None) -> List[tupl
 # Usage example:
 if __name__ == "__main__":
     embedder = Embedder()
-    doc = ["troll vn troll vn troll vn"]
+    doc = "troll vn troll vn troll vn"
     doc_embedding = embedder.embed(doc)
     print(doc_embedding.shape)
```
Comment (khankaholic) [inline app/llama_sensei/backend/add_courses/embedding/get_embedding.py]: @micache 
consider to remove workaround code

File: app/llama_sensei/backend/qa/generate_answer.py [hunk 2]
```diff
@@ -1,31 +1,48 @@
...
     DocumentProcessor,
 )
 from ragas import evaluate
 from ragas.metrics import faithfulness
+from sentence_transformers import SentenceTransformer
+from sklearn.metrics.pairwise import cosine_similarity
 
 MODEL = "llama3-70b-8192"
+# device = "cuda" if torch.cuda.is_available() else "cpu"
+# EMBEDDING_LLM = SentenceTransformer("all-MiniLM-L12-v2", trust_remote_code=True).to(device)
```
Comment (khankaholic) [inline app/llama_sensei/backend/qa/generate_answer.py]: consider to remove comment
we do have history to trace back, no need to do that

File: app/llama_sensei/backend/qa/generate_answer.py [hunk 3]
```diff
@@ -1,31 +1,48 @@
...
+EMBEDDING_LLM = "all-MiniLM-L12-v2"
 
 
 class GenerateRAGAnswer:
     def __init__(self, query: str, course: str, model=MODEL):
         self.query = query
         self.course = course
+        device = "cuda" if torch.cuda.is_available() else "cpu"
```
Comment (khankaholic) [inline app/llama_sensei/backend/qa/generate_answer.py]: `device` has no duty in initialization here, move it out just like `EMBEDDING_LLM`

File: app/llama_sensei/backend/qa/generate_answer.py [hunk 4]
```diff
@@ -1,31 +1,48 @@
...
+        )
         self.model = ChatGroq(model=model, temperature=0)
-        self.contexts = None  # To store the retrieved contexts
+        self.contexts = []  # To store the retrieved contexts
 
     def retrieve_contexts(self):
         processor = DocumentProcessor(self.course, search_only=True)
         result = processor.search(self.query)
+
         # print(result)
```
Comment (khankaholic) [inline app/llama_sensei/backend/qa/generate_answer.py]: remove comment code, same reason as above

File: app/llama_sensei/backend/qa/generate_answer.py [hunk 5]
```diff
@@ -107,44 +118,98 @@ def calculate_faithfulness(self, generated_answer: str) -> float:
...
     # Create an instance of GenerateRAGAnswer with the query and course
-    rag_generator = GenerateRAGAnswer(query=prompt, course=course)
+    rag_generator = GenerateRAGAnswer(query=prompt, course=course_name)
 
     # Generate and print the answer along with its embedded faithfulness score
-    answer = rag_generator.generate_answer()
-    print(answer)
+    answer, evidence = rag_generator.generate_answer()
+    print("answer:\n" + answer)
+    # print(evidence)
```
Comment (khankaholic) [inline app/llama_sensei/backend/qa/generate_answer.py]: consider to remove workaround code

File: app/llama_sensei/backend/qa/generate_answer.py
Comment (khankaholic) [inline app/llama_sensei/backend/qa/generate_answer.py:1]: maybe you guys can rethink about how do you want to use this class.
I see a problem that is whenever I need an answer of a specific course, I need to create an instance of this class again to pass the query into it and that's very exhausted.

I'm considering this, please take it into account:
```python
rag_generator = GenerateRAGAnswer(course=course_name)
query = "hello croissant"
answer = rag_generator.generate_answer(query=query)
```
and even if I want to change the course to utilize existed instance, it should go like this using `@property` and `setter`:
```python
rag_generator.course = new_course
```

File: app/llama_sensei/backend/qa/generate_answer.py [hunk 6]
```diff
@@ -53,44 +70,38 @@ def gen_prompt(self) -> str:
...
+        results = wrapper.results(self.query, max_results=5)
 
-        before = datetime.now()
-        res = self.model.invoke(final_prompt)
-        print(f"LLM return time: {datetime.now() - before} seconds")
+        return results
 
-        llm_answer = res['content'] if isinstance(res, dict) else res.content
+    def calculate_context_relevancy(self) -> float:
+        # Embed the query]
```
Comment (khankaholic) [inline app/llama_sensei/backend/qa/generate_answer.py]: typo of comment

**Review by khankaholic — CHANGES_REQUESTED** [review]

**Review by khankaholic — CHANGES_REQUESTED** [review]
pre-commit has failed. let's make it alive again guys @micache @thaohp113

File: app/llama_sensei/backend/qa/generate_answer.py
Comment (micache) [reply app/llama_sensei/backend/qa/generate_answer.py:1]: ok i fixed

File: app/llama_sensei/frontend/pages/1_upload.py [hunk 7]
```diff
@@ -82,4 +81,5 @@ def upload():
     shutil.rmtree(os.path.join("data", course_name))
     shutil.rmtree(os.path.join("data/transcript", course_name))
```
Comment (Huy1711) [inline app/llama_sensei/frontend/pages/1_upload.py:82]: no need to remove upload audio/transcript data, please remove these 2 lines

**Review by Huy1711 — APPROVED** [review]
